nilearn/region_extraction/SimpleThresholdExtractor.py

```diff
@@ -0,0 +1,139 @@
+
+"""
+Simple Threshold Extractor
+"""
+
+import nibabel
+import numpy as np
+
+from scipy.ndimage import label
+from scipy.stats import scoreatpercentile
+
+from sklearn.base import BaseEstimator
+
+from nilearn import plotting
+from nilearn._utils import check_niimg
+from nilearn.image import iter_img
+
+
+def extract_connected_components(img, min_size):
+    """ Extract the components which have largest
+        connections given a thresholded image of the
+        components
+
+    Parameters
+    ----------
+    img: 3D/4D Nifti like object
+        a thresholded image of the components
+
+    Returns
+    -------
+    the extracted largest connected components of each thresholded
+    map
+
+    """
+    components, labels = label(img)
+    label_count = np.bincount(components.ravel())
+    label_count[0] = 0
+    for label_assign, count in enumerate(label_count):
+        if count < min_size:
+            img[components == label_assign] = 0
```

Comment:
You should actually extract the regions, not set the ones that you don't want to 0. Meaning, if I give you a 3D map with 4 regions in it, you should output a 4D file with 4 scans and I should find one region per scan.